l10n_it_edi_extension/models/account_move.py

```diff
@@ -0,0 +1,874 @@
+                        ),
+                        "invoice_id": self.id,
+                    }
+                )
+                self.l10n_it_edi_amount_tax += get_float(element_summary, ".//Imposta")
+
+        extra_info["l10n_it_edi_ext_body_tree"] = body_tree
+        return extra_info, message_to_log
+
+    def _l10n_it_edi_create_partner(
+        self, xml_tree, partner_section_xpath, vat, codice_fiscale
+    ):
+        country_id = False
+        is_company = bool(get_text(xml_tree, partner_section_xpath + "//Denominazione"))
+        eori_code = get_text(xml_tree, partner_section_xpath + "//CodEORI")
+
+        if country_code := get_text(xml_tree, partner_section_xpath + "//IdPaese"):
+            countries = self.env["res.country"].search([("code", "=", country_code)])
+            if countries:
+                country_id = fields.first(countries).id
+            else:
+                raise UserError(
+                    self.env._("Country Code %s not found in system.") % country_code
+                )
+
+        vals = {
+            "vat": country_code + vat,
+            "l10n_it_codice_fiscale": codice_fiscale,
+            "is_company": is_company,
+            "l10n_edi_it_eori_code": eori_code,
+            "country_id": country_id,
+        }
+
+        if value := get_text(xml_tree, partner_section_xpath + "//Denominazione"):
+            vals["name"] = value
+        else:
+            vals["name"] = " ".join(
+                filter(
+                    None,
+                    [
+                        get_text(xml_tree, partner_section_xpath + "//Nome"),
+                        get_text(xml_tree, partner_section_xpath + "//Cognome"),
+                    ],
+                )
+            )
+
+        return self.env["res.partner"].create(vals)
+
+    def _l10n_it_edi_update_partner(self, xml_tree, partner_section_xpath, partner):
+        vals = {}
+
+        address_parts = filter(
+            None,
+            [
+                get_text(xml_tree, partner_section_xpath + "//Indirizzo"),
+                get_text(xml_tree, partner_section_xpath + "//NumeroCivico"),
+            ],
+        )
+        vals["street"] = " ".join(address_parts)
+
+        for field_name, xml_path in [
+            ("zip", "//CAP"),
+            ("city", "//Comune"),
+            ("l10n_edi_it_register", "//AlboProfessionale"),
+            ("phone", "//Telefono"),
+            ("email", "//Email"),
+            ("l10n_edi_it_register_code", "//NumeroIscrizioneAlbo"),
+        ]:
+            value = get_text(xml_tree, partner_section_xpath + xml_path)
+            vals[field_name] = value
+
+        if province := get_text(xml_tree, partner_section_xpath + "//Provincia"):
+            if provinces := self.env["res.country.state"].search(
+                [("code", "=", province), ("country_id", "=", partner.country_id.id)]
+            ):
+                vals["state_id"] = fields.first(provinces).id
```

Comment:
```suggestion
            if province := self.env["res.country.state"].search(
                [("code", "=", province), ("country_id", "=", partner.country_id.id)], limit=1
            ):
                vals["state_id"] = province.id
```

Reply:
per @monen17

Reply:
Aggiunto in https://github.com/Borruso/l10n-italy/pull/33